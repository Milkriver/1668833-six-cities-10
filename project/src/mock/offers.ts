import { Offers } from '../types/offer';

export const offers:Offers = [
  {
    id: 1,
    image: ['img/room.jpg', 'img/apartment-01.jpg', 'img/apartment-02.jpg', 'img/apartment-03.jpg', 'img/studio-01.jpg', 'img/apartment-01.jpg'],
    premium: true,
    bookmark: false,
    name: 'Beautiful & luxurious apartment at great location',
    type: 'Apartment',
    price: 120,
    rating: 4.8,
    bedrooms: 3,
    guests: 4,
    options: ['Wi-Fi', 'Washing machine', 'Towels', 'Heating', 'Coffee machine', 'Baby seat', 'Kitchen', 'Dishwasher', 'Cabel TV', 'Fridge'],
    host: {
      image: 'img/avatar-angelina.jpg',
      name: 'Angelina',
      status: 'Pro',
    },
    description: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.The building is green and from 18th century. An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.',
    reviews: [
      {
        avatar: 'img/avatar-max.jpg',
        name: 'Max',
        text: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.The building is green and from 18th century.',
        date: '2019-04-24',
        rating: 4,
      }
    ],
  },
  {
    id: 2,
    image: ['img/room.jpg', 'img/apartment-01.jpg', 'img/apartment-02.jpg', 'img/apartment-03.jpg', 'img/apartment-01.jpg'],
    premium: false,
    bookmark: true,
    name: 'Wood and stone place',
    type: 'Private room',
    price: 80,
    rating: 4,
    bedrooms: 3,
    guests: 4,
    options: ['Wi-Fi', 'Heating', 'Coffee machine', 'Kitchen', 'Dishwasher', 'Cabel TV', 'Fridge'],
    host: {
      image: 'img/avatar-angelina.jpg',
      name: 'Angelina',
      status: 'Pro',
    },
    description: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.The building is green and from 18th century. An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.',
    reviews: [
      {
        avatar: 'img/avatar-max.jpg',
        name: 'Max',
        text: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.The building is green and from 18th century.',
        date: '2019-04-24',
        rating: 4,
      }
    ],
  },
  {
    id: 3,
    image: ['img/apartment-02.jpg', 'img/apartment-03.jpg', 'img/apartment-01.jpg'],
    premium: false,
    bookmark: false,
    name: 'Canal View Prinsengracht',
    type: 'Apartment',
    price: 132,
    rating: 4,
    bedrooms: 3,
    guests: 4,
    options: ['Wi-Fi', 'Heating', 'Coffee machine', 'Kitchen', 'Dishwasher', 'Cabel TV', 'Fridge'],
    host: {
      image: 'img/avatar-angelina.jpg',
      name: 'Angelina',
      status: 'Pro',
    },
    description: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.The building is green and from 18th century. An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.',
    reviews: [
      {
        avatar: 'img/avatar-max.jpg',
        name: 'Max',
        text: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.The building is green and from 18th century.',
        date: '2019-04-24',
        rating: 4,
      }
    ],
  },
  {
    id: 4,
    image: ['img/apartment-03.jpg', 'img/apartment-01.jpg'],
    premium: false,
    bookmark: false,
    name: 'Nice, cozy, warm big bed apartment',
    type: 'Apartment',
    price: 125,
    rating: 4,
    bedrooms: 2,
    guests: 2,
    options: ['Wi-Fi', 'Coffee machine', 'Kitchen', 'Dishwasher', 'Cabel TV', 'Fridge'],
    host: {
      image: 'img/avatar-angelina.jpg',
      name: 'Helen',
      status: 'Pro',
    },
    description: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.The building is green and from 18th century. An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.',
    reviews: [
      {
        avatar: 'img/avatar-max.jpg',
        name: 'Max',
        text: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.The building is green and from 18th century.',
        date: '2019-04-24',
        rating: 4,
      }
    ],
  },
];
