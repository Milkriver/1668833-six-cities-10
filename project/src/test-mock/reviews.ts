export const mockReviews = [
  {
    comment: 'The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.',
    date: '2022-06-23T12:25:36.939Z',
    id: 1,
    rating: 3,
    user: {
      avatarUrl: 'https://10.react.pages.academy/static/avatar/3.jpg',
      id: 12,
      isPro: true,
      name: 'Isaac'
    }
  },
  {
    comment: 'I stayed here for one night and it was an unpleasant experience.',
    date: '2022-06-23T12:25:36.939Z',
    id: 2,
    rating: 5,
    user: {
      avatarUrl: 'https://10.react.pages.academy/static/avatar/2.jpg',
      id: 11,
      isPro: false,
      name: 'Jack'
    }
  }
];
