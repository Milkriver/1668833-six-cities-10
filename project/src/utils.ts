import { City, Offer } from './types/offer';

export const sortCityOffers = (offers: Offer[], city: string) => (
  offers.filter((offer) => offer.city.name === city)
);

export const changeCity = (locations: City[], city: string) => (
  locations.find((location) => location.name === city)
);
