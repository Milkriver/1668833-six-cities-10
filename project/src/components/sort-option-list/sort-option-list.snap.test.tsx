import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { configureMockStore } from '@jedmao/redux-mock-store';
import { Provider } from 'react-redux';
import HistoryRouter from '../../components/history-router/history-router';
import SortOptionList from './sort-option-list';
import { mockOffers } from '../../test-mock/offers';
import { sortCityOffers } from '../../utils';

const mockStore = configureMockStore();

describe('Component: SortOptionList', () => {
  it('should render correctly', () => {
    const history = createMemoryHistory();
    const store = mockStore({
      DATA: {
        selectedSortOption: 'Popular'
      }
    });
    const cityOffers = sortCityOffers(mockOffers, mockOffers[0].city.name);
    const { container } = render(
      <Provider store={store}>
        <HistoryRouter history={history}>
          <SortOptionList offers={cityOffers} />
        </HistoryRouter>
      </Provider>);
    expect(container).toMatchSnapshot();
  });
});
