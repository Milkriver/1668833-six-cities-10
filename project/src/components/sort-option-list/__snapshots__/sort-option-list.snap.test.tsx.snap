// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SortOptionList should render correctly 1`] = `
<div>
  <form
    action="#"
    class="places__sorting"
    method="get"
  >
    <span
      class="places__sorting-caption"
    >
      Sort by
    </span>
    <span
      class="places__sorting-type"
    >
      Popular
      <svg
        class="places__sorting-arrow"
        height="4"
        width="7"
      >
        <use
          xlink:href="#icon-arrow-select"
        />
      </svg>
    </span>
  </form>
</div>
`;
